Content.Server/Physics/Components/RandomWalkComponent.cs

```diff
@@ -0,0 +1,52 @@
+using Content.Server.Physics.Controllers;
+
+namespace Content.Server.Physics.Components;
+
+/// <summary>
+/// A component which makes its entity move around at random.
+/// </summary>
+[RegisterComponent]
+public sealed class RandomWalkComponent : Component
+{
+    /// <summary>
+    /// The minimum speed at which this entity will move.
+    /// </summary>
+    [DataField("minSpeed")]
+    [ViewVariables(VVAccess.ReadWrite)]
+    public float MinSpeed = 7.5f;
+
+    /// <summary>
+    /// The maximum speed at which this entity will move.
+    /// </summary>
+    [DataField("maxSpeed")]
+    [ViewVariables(VVAccess.ReadWrite)]
+    public float MaxSpeed = 10f;
+
+    /// <summary>
+    /// The amount of speed carried over when the speed updates.
+    /// </summary>
+    [DataField("accumulatorRatio")]
+    [ViewVariables(VVAccess.ReadWrite)]
+    public float AccumulatorRatio = 0.0f;
```

Comment:
Alright, I'll get on that in the morning.

Reply:
> I reccomend you use timespan targets instead of accumulators
Also, you're looking at the wrong type of accumulator there. That's how much of the old velocity gets carried over when the entity gets punted in a new direction.

Reply:
Oh sorry, instantly assumed it was an accumulator, you cant resolve this if there arent any

Reply:
That should do it.